cmd/kube-apiserver/app/server.go

```diff
@@ -393,6 +410,14 @@ func (s *APIServer) Run(_ []string) error {
 		}
 	}
 
+	databaseStorage := master.DatabaseClients{
+		Default:      etcdStorage,
+		Experimental: expEtcdStorage,
+	}
+	if etcdEventStorage != nil {
+		databaseStorage.Overwrites = map[string]storage.Interface{"events": etcdEventStorage}
```

Comment:
I suggest you fully specify the group & resource in this map. (Presumably you want the same storage for multiple versions, so you can leave version out.)

e.g. make the map key a `struct { GroupName string; ResourceName string }`